sklearn/neighbors/_subsampled.py

```diff
@@ -0,0 +1,193 @@
+"""Subsampled neighbors transformer"""
+
+# Author: Jennifer Jang <j.jang42@gmail.com>
+#         Heinrich Jiang <heinrichj@google.com>
+#
+# License: BSD 3 clause
+
+import numpy as np
+
+from ..metrics.pairwise import paired_distances, PAIRED_DISTANCES
+from ._base import UnsupervisedMixin
+from ..base import TransformerMixin, BaseEstimator
+from ..utils import check_random_state
+from ..utils.validation import check_is_fitted, _deprecate_positional_args
+from ..utils.validation import check_array
+
+
+class SubsampledNeighborsTransformer(TransformerMixin, UnsupervisedMixin, 
+    BaseEstimator):
+    """Compute subsampled sparse distance matrix of neighboring points in X.
+
+    Parameters
+    ----------
+
+    s : float
+        Sampling probability.
+
+    eps : float, default=None
+        Neighborhood radius. Pairs of points which are at most eps apart are 
+        considered neighbors. If None, radius is assumed to be infinity.
+
+    metric : string or callable, default='euclidean'
+        Input to paired_distances function. Can be string specified 
+        in PAIRED_DISTANCES, including "euclidean", "manhattan", or 
+        "cosine." Alternatively, can be a callable function, which should 
+        take two arrays from X as input and return a value indicating 
+        the distance between them.
+
+    random_state : int, RandomState instance, default=None
+        Seeds the random sampling of lists of vertices. Use an int to 
+        make the randomness deterministic.
+        See :term:`Glossary <random_state>`.
+
+    References
+    ----------
+    - Faster DBSCAN via subsampled similarity queries, 2020
+        Heinrich Jiang, Jennifer Jang, Jakub Łącki
+        https://arxiv.org/abs/2006.06743
+
+    Notes
+    -----
+    Each pair of points in X is sampled uniformly with probability s, 
+    and the final distance matrix is symmetric.
+    """
+
+    @_deprecate_positional_args
+    def __init__(self, s, eps=None, *, metric='euclidean', random_state=None):
+        self.s = s
+        self.eps = eps
+        self.metric = metric
+        self.random_state = random_state
+
+
+    def _fit(self, X):
```

Comment:
```suggestion
    def _check_parameters(self, X):
```

Reply:
I moved the checking to a **_check_parameters** function.